includes/payment-methods/class-wc-stripe-upe-payment-method-bacs-debit.php

```diff
@@ -85,4 +92,23 @@ public function create_payment_token_for_user( $user_id, $payment_method ) {
 		$token->save();
 		return $token;
 	}
+
+	public function hide_bacs_for_subscriptions_with_free_trials() {
+		add_filter(
+			'woocommerce_available_payment_gateways',
+			function ( $available_gateways ) {
+				global $post;
+				$is_checkout_shortcode_page          = wc_post_content_has_shortcode( 'woocommerce_checkout' ) || has_block( 'woocommerce/classic-shortcode', $post );
+				$is_update_order_review_ajax_request = defined( 'DOING_AJAX' ) && DOING_AJAX && isset( $_REQUEST['wc-ajax'] ) && 'update_order_review' === $_REQUEST['wc-ajax'];
```

Comment:
Should we disable `phpcs` here?
```php
// phpcs:disable WordPress.Security.NonceVerification.Recommended
```

Reply:
In this case, it's mainly my IDE that's highlighting the issue (see image).

![image](https://github.com/user-attachments/assets/233b7616-c349-4098-b4b1-9071d442a023)

However, it's not a blocker. So, feel free to leave as-is.